crates/wdk-build/src/cargo_make.rs

```diff
@@ -566,13 +639,61 @@ pub fn copy_to_driver_package_folder<P: AsRef<Path>>(path_to_copy: P) -> Result<
 /// - [`ConfigError::MultipleWDKBuildCratesDetected`] if there are multiple
 ///   versions of the WDK build crate detected
 /// - [`ConfigError::IoError`] if there is an error creating or updating the
-///   symlink to `rust-driver-toolchain.toml`
+///   symlink to `rust-driver-makefile.toml`
 ///
 /// # Panics
 ///
 /// This function will panic if the `CARGO_MAKE_WORKSPACE_WORKING_DIRECTORY`
 /// environment variable is not set
 pub fn load_rust_driver_makefile() -> Result<(), ConfigError> {
+    load_rust_driver_makefile_internal(RUST_DRIVER_MAKEFILE_NAME)
+}
+
+/// Symlinks `rust-driversample-makefiletoml` to the `target` folder where it
+/// can be extended from a `Makefile.toml`. This is necessary so that paths in
+/// the `rust-driver-sample-makefile.toml` can to be relative to
+/// `CARGO_MAKE_CURRENT_TASK_INITIAL_MAKEFILE_DIRECTORY`
+///
+/// # Errors
+///
+/// This function returns:
+/// - [`ConfigError::CargoMetadataError`] if there is an error executing or
+///   parsing `cargo_metadata`
+/// - [`ConfigError::MultipleWDKBuildCratesDetected`] if there are multiple
+///   versions of the WDK build crate detected
+/// - [`ConfigError::IoError`] if there is an error creating or updating the
+///   symlink to `rust-driver-sample-makefile.toml`
+///
+/// # Panics
+///
+/// This function will panic if the `CARGO_MAKE_WORKSPACE_WORKING_DIRECTORY`
+/// environment variable is not set
+pub fn load_rust_driver_sample_makefile() -> Result<(), ConfigError> {
+    load_rust_driver_makefile_internal(RUST_DRIVER_SAMPLE_MAKEFILE_NAME)
+}
+
+/// Symlinks a given Windows Driver rust makefile to the `target` folder where
+/// it can be extended from a `Makefile.toml`. This is necessary so that paths
+/// in the Windows Driver makefile can to be relative to
+/// `CARGO_MAKE_CURRENT_TASK_INITIAL_MAKEFILE_DIRECTORY`
+///
+/// # Errors
+///
+/// This function returns:
+/// - [`ConfigError::CargoMetadataError`] if there is an error executing or
+///   parsing `cargo_metadata`
+/// - [`ConfigError::MultipleWDKBuildCratesDetected`] if there are multiple
+///   versions of the WDK build crate detected
+/// - [`ConfigError::IoError`] if there is an error creating or updating the
+///   symlink to the makefile.
+///
+/// # Panics
+///
+/// This function will panic if the `CARGO_MAKE_WORKSPACE_WORKING_DIRECTORY`
+/// environment variable is not set
+fn load_rust_driver_makefile_internal<S: AsRef<str> + AsRef<Utf8Path> + AsRef<Path>>(
```

Comment:
I think you can remove the traitbound on `AsRef<Path>>`

Reply:
You'd think so, but the compiler is unhappy when I try.